cmd/frontend/internal/pkg/handlerutil/middleware.go

```diff
@@ -35,6 +36,13 @@ func CSRFMiddleware(next http.Handler, isSecure func() bool) http.Handler {
 	v.Store(newHandler(isSecure()))
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// 🚨 SECURITY: this is to let POST and PUT requests through that are intended for the grafana instance
+		// running behind a reverse proxy (see cmd/frontend/internal/app/debug.go).
+		// this is ok because it is contained to grafana and grafana itself uses cookies with SameSite=lax attribute
```

Comment:
```suggestion
		// this is ok because it is contained to grafana and grafana itself uses CSRF cookies
		// with SameSite=lax attribute
```

Is it more accurate to include the word CSRF here?

Reply:
AFA we investigated, Grafana does not use CSRF to protect/validate but rely on SameSite cookie attribute. That's why we couldn't configure it to carry our `sg_xsrf_token` and need this whitelist.

Ref: https://github.com/grafana/grafana/issues/1441